src/common/hooks/useFeaturedPlays.js

```diff
@@ -1,33 +1,87 @@
 import { useEffect, useState } from "react";
 import { submit } from "common/services/request";
-import { FetchPlaysFilter } from "common/services/request/query/fetch-plays-filter";
-
-/**
- * run graphql query to retrive featured plays
- * @returns [loading, error, data]
- */
+import { FetchPlaysSimple } from "common/services/request/query/fetch-plays";
 
 const useFeaturedPlays = () => {
-  const { getAllFeaturedPlays } = FetchPlaysFilter;
+  const [viewsData, setViewsData] = useState([]);
   const [data, setData] = useState([]);
   const [error, setError] = useState(null);
   const [loading, setLoading] = useState(false);
+  const [slugs, setSlug] = useState([]);
+  const [trendingPlays, setTrendingPlays] = useState([]);
+
+  async function fetchTrendingPlays(start, end, headers) {
+    return await fetch(
+      `${process.env.REACT_APP_UMAMI_METRICS_URL}?start_at=${start}&end_at=${end}&type=url`,
+      { headers }
+    );
+  }
 
   useEffect(() => {
+    function slugFromViewsData(slug) {
+      return setSlug((oldVal) => [...oldVal, slug.split("/")[3]]);
+    }
+
+    const headers = {
+      Authorization: `Bearer ${process.env.REACT_APP_UMAMI_BEARER_TOKEN}`,
+    };
+    const weeklyStartTime = 1661365800000;
+    const weeklyEndTime = 1661797800000;
+
+    const unwantedPaths = [
+      "/",
+      "/plays",
+      "/ideas",
+      "/plays/",
+      "/?ref=madewithreactjs.com",
+      "/tech-stacks",
+      "/?error=invalid-request&errorDescription=%22redirectTo%22%20does%20not%20match%20any%20of%20the%20allowed%20types",
+      "/ideas/",
+      "/plays/create",
+      "/plays/98415e0a-68d2-4930-a7ff-3559200740fe",
+      "/plays/editplay",
+    ];
+
     (async () => {
       try {
         setLoading(true);
-        const res = await submit(getAllFeaturedPlays());
-        setData(res);
+        const viewsRes = await fetchTrendingPlays(
+          weeklyStartTime,
+          weeklyEndTime,
+          headers
+        );
+        console.log(viewsRes);
+        setViewsData(viewsRes);
+
+        setViewsData((current) =>
+          current.filter((d) => {
+            return unwantedPaths.includes(d.x) === false;
+          })
+        );
+        viewsData.slice(0, 4).map((v) => {
+          return slugFromViewsData(v.x);
+        });
+        console.log(slugs);
+        const dataRes = await submit(FetchPlaysSimple());
+        setData(dataRes);
+        setData((current) =>
+          current.filter((d) => {
```

Comment:
this could be written as 
`
current.filter((d)=>viewsData.filter((v)=>v.x ===d.slug))
`